NexusMods.Abstractions.GC.DataModel/BackedUpFile.cs

```diff
@@ -0,0 +1,21 @@
+using JetBrains.Annotations;
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+namespace NexusMods.Abstractions.GC.DataModel;
+
+/// <summary>
+/// Represents a file which is stored as part of a backup.
+/// This is a file that forms a GC root.
+///
+/// Specific types of backed-up files will be derived from this class.
+/// </summary>
+[PublicAPI]
+public partial class BackedUpFile : IModelDefinition
+{
+    private const string Namespace = "NexusMods.GC.BackedUpFile";
+
+    /// <summary>
+    /// Hash of the file.
+    /// </summary>
+    public static readonly HashAttribute Hash = new(Namespace, nameof(Hash)) { IsIndexed = true };
+}
```

Comment:
I think this should go in `Abstractions.Loadout` and we can avoid the `Abstractions.GC.DataModel` project.

Reply:
Use single level without inheritance, with both hash and gameMetadata